mlir/lib/Dialect/MIOpen/Generator/Conv2dGenerator.cpp

```diff
@@ -96,6 +100,15 @@ LogicalResult hasDimensions(const llvm::StringMap<int64_t> &map,
   return success();
 }
 
+LogicalResult smallEnough(const ArrayRef<int64_t> dims, size_t elemWidth, StringRef name) {
+  int64_t size = std::accumulate(dims.begin(), dims.end(), 1LL,
+    std::multiplies<int64_t>()) * elemWidth;
+  if (size >= (1LL << 31)) { // 2^31 = 2 GB
+    llvm::errs() << name << " tensor cannot be larger than 2 GB\n";
```

Comment:
Recoverable error should not go with `llvm::errs()`. Probably use `LLVM_DEBUG`.

Reply:
It's not recoverable? We can't generate code with tensors of that size

Reply:
From the aspect of the client, this is part of applicability check. If it's not-applicable client will actually not chase it down. But it will be really confusing for the client to see an error message from us while everything is fine.

Reply:
Fair enough, putting that through to debug is probably the right place then.

Reply:
Fixed in https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/pull/381/commits/dd1fcb6ce83199a7a35c0b548f3333204d3b86a7

However, one item of note: when llvm is built in Release mode or its variants (as we'll probably be doing for shipping to downstream) llvm::dbgs() == llvm::errs()